### PR TITLE
Add rerouting-container-registries-with-envoy post

### PR DESCRIPTION
**Pull request by BobyMCbobs**

### CODE REVIEW

**Review by hh — APPROVED** [review]
/lgtm

**Review by zachmandeville — COMMENTED** [review]
loooookin' good, just a couple of changes!

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+In this post, I will detail the discovery of Envoy's dynamic rewriting location capabilities and the relationship to OCI registries.
+
+**What is [[https://www.envoyproxy.io/][Envoy]]?**
+  #+BEGIN_QUOTE
+  open source edge and service proxy, designed for cloud-native applications
+  #+END_QUOTE
+
+**What is an [[https://opencontainers.org/][OCI container registry]]?**
+  #+BEGIN_QUOTE
+  a standard and specification for implementation container registries
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:11]: This is good

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  #+END_QUOTE
+
+I've been playing around with and learning Envoy for a number of months now. One of the concepts I'm investigating is rewriting the request's host.
+Envoy is a super powerful piece of software. It is flexible and highly dynamic.
+
+* Dependencies
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:17]: `sudo apt uninstall -y envoyproxy`

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+You'll need:
+- [[https://www.envoyproxy.io/docs/envoy/latest/start/install][envoy]]
+
+* Journey
+** My expectations
+The goal is to set up Envoy on a host to rewrite all requests dynamically back to a container registry hosted by a cloud-provider, such as GCP.
+
+** Initial discoveries
+One of the first things I investigated was the ability to get traffic from one site and serve it on another (proxying).
+I searched forth in the docs and found in the [[https://www.envoyproxy.io/docs/envoy/v1.17.1/start/quick-start/configuration-static][most basic example]] that it's possible to use Envoy's http filter in the /filter_chains/ a static host is able to be rewritten.
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:27]: Sounds better to me

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+This is a great start! This serves the site and it's content under the host where Envoy is served.
+However, the host in the rewrite is static and not dynamic. It seems at this point like doing the implementation this way is not viable.
+
+** Learning about filter-chains
+Envoy has the lovely feature to set many kinds of middleware in the middle of a request.
+This middleware can be used to add/change/remove things from the request.
+The functionality is infinitely useful as filters can be such things as gRPC, PostgreSQL, Wasm, so on.
+
+Envoy is particularly good at HTTP related filtering. It also supports such features as dynamic forward proxy, JWT auth, health checks, rate limiting, and Lua.
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:74]: I put Lua as the last thing to lead onto the next section, I'll rearrange things

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: ii.coop
+                port_value: 443
+#+END_SRC
+
+
+With envoy running this config, the behaviour of the requests is:
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:151]: Makes more sense, I like it

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+rewrite all traffic hitting the web service to /k8s.gcr.io/, except if the IP is /192.168.0.1/ then set the location to /registry-1.docker.io/.
+
+Since I'm using a [[https://github.com/sharingio/pair][Pair]] instance, it set's the local subnet to /192.168.0.0/24/ so when I try to =docker pull humacs-envoy-10000.$SHARINGIO_PAIR_BASE_DNS_NAME/library/postgres:12-alpine= it will go to /docker.io/.
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:155]: Possessive or not possessive that is the question

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+On my local machine, pulling container images using =docker pull humacs-envoy-10000.$SHARINGIO_PAIR_BASE_DNS_NAME/e2e-test-images/agnhost:2.26= will instead use /k8s.gcr.io/.
+
+To achieve this, I research how other http libraries handle redirects - namely [[https://golang.org/src/net/http/server.go?s=66471:66536#L2179][Golang's net/http.Redirect]].
+The main things that Golang's /http.Redirect/ does it:
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org:160]: Whoopseeees

File: blog/org/rerouting-container-registries-with-envoy.org [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+- set the /content-type/ header to /text/html/
+- set the location to the destination
+- set the status code to 302
+- set the body to the same data in earlier steps, but in an /a/ tag.
+
+* Final thoughts
+I'm learning that Envoy is highly flexible and seemly limitless in it's capabilities.
+It's exciting to see Envoy being adopted in many places, it's usecases, and implementations.
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply blog/org/rerouting-container-registries-with-envoy.org]: I like it

**Pull request merged**